client/src/api/v2/camera.js

```diff
@@ -39,3 +41,140 @@ export function useOverlays(device) {
+ * Send a message to server to stop recording video
+ */
+export function stopRecording() {
+  emit('stop-camera-recording');
+}
+
+/**
+ * Get the last MQTT payload for the status of a camera recording
+ *
+ * @param {string} device Camera
+ * @returns {string} Payload
+ */
+export function getLastPayload(device) {
+  return sessionStorage.getItem(`camera-recording-last-payload-${device}`);
+}
+
+/**
+ * Store the last MQTT payload for the status of a camera recording
+ *
+ * @param {string} device Camera
+ * @param {string} payloadString String representation of payload
+ * @returns {string} Payload
+ */
+export function storeLastPayload(device, payloadString) {
+  return sessionStorage.setItem(`camera-recording-last-payload-${device}`, payloadString);
+}
+
+/**
+ * Parse the payload to get the status
+ *
+ * @param {string} payload Payload
+ * @returns {string} Status
+ */
+export function getStatusFromPayload(payload) {
+  const data = JSON.parse(payload);
+  if (!data) return null;
+  return data.status;
+}
+
+/**
+ * Parse the payload to get the information (fields other than status)
+ *
+ * Payload structure is defined in the 'V3 MQTT Topics' page on Notion.
+ * Topic is /v3/camera/recording/status/<primary/>secondary>.
+ *
+ * @param {string} payload Payload
+ * @returns {React.Component} Component
+ */
+export function getInfoFromPayload(payload) {
+  const data = JSON.parse(payload);
+  if (!data) return null;
+
+  const format = (field) => {
+    let name = startCase(field);
+    let value = '';
+
+    // format field value
+    switch (field) {
+      case 'diskSpaceRemaining':
+        value = formatBytes(data.diskSpaceRemaining);
+        break;
+
+      case 'recordingMinutes':
+        {
+          const mins = Math.floor(data.recordingMinutes);
+          const secs = (data.recordingMinutes - mins) * 60;
+          value = `${mins} minutes ${secs} seconds`;
+          name = 'Recording Time';
+        }
+        break;
+
+      default:
+        value = data[field];
+        break;
+    }
+
+    return `${name}: ${value}`;
+  };
+
+  return (
+    <>
+      {
+        Object.keys(data)
+          .filter((field) => field !== 'status')
+          .map((field) => (
+            <p>{ format(field) }</p>
+          ))
+      }
+    </>
+  );
+}
+
+/**
+ * @typedef {object} StatusPayload
+ * @property {string} status Status of payload
+ * @property {string} payload Last payload published
+ */
+
+/**
+ * Returns the status of the camera and the last payload published
+ *
+ * @param {string} device Device
+ * @returns {StatusPayload} Status and payload
+ */
+export function useCameraRecordingStatus(device) {
+  const [lastPayload, setLastPayload] = useState(getLastPayload(device));
+  const [status, setStatus] = useState(getStatusFromPayload(lastPayload) || 'Waiting for status...');
+
+  const update = useCallback((newPayload) => {
+    // update last payload
+    storeLastPayload(device, newPayload);
+    setLastPayload(newPayload);
```

Comment:
Having trouble getting my head around this, what's the difference between `storeLastPayload` and `setLastPayload` here?

Reply:
Also, should we be storing the most recent statuses from the cameras serverside? Because the most recent status in the `sessionStorage` isn't necessarily the most up to date, there could have been more statuses while the website was not open.

Reply:
> Having trouble getting my head around this, what's the difference between `storeLastPayload` and `setLastPayload` here?
`storeLastPayload` has been removed as it is now handled server-side. `setLastPayload` is used to update the state of `lastPayload`.

4376700